Userland/Libraries/LibChess/Chess.cpp

```diff
@@ -774,9 +774,11 @@ Board::Result Board::game_result() const
     });
 
     if (are_legal_moves) {
-        if (m_moves_since_capture >= 75 * 2)
+        if (m_moves_since_capture >= 75 * 2 && m_moves_since_pawn_advance >= 75 * 2)
             return Result::SeventyFiveMoveRule;
-        if (m_moves_since_capture == 50 * 2)
+
+        if ((m_moves_since_capture >= 50 * 2 && m_moves_since_pawn_advance == 50 * 2)
+            || (m_moves_since_pawn_advance >= 50 * 2 && m_moves_since_capture == 50 * 2))
```

Comment:
Why is this written like that and not just check them like for the `SeventyFiveMoveRule`?

Reply:
I've written it so that the 50 move rule is only triggered on move 50. The game can continue if the player decides not to accept a draw at that . If it were written like the 75 move rule above, a dialog box would appear on every move between move 50 and 75 asking if the player would like a draw. While I think this would be technically correct, it is annoying.

The 75 move rule can be written more concisely, as the game shouldn't continue after that condition is reached.

Reply:
I could image setting a flag here if the draw is declined. But lets see what our maintainers think about that.

Reply:
I considered doing the same thing, but thought it would be better to keep the logic as close as possible to what was there previously.